src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -32,10 +35,10 @@ protected function configure()
     {
         $this
             ->setName('security:encode-password')
-            ->setDescription('Encode a password.')
+            ->setDescription('Encodes a password.')
             ->addArgument('password', InputArgument::OPTIONAL, 'Enter a password')
-            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.')
-            ->addArgument('salt', InputArgument::OPTIONAL, 'Enter the salt you want to use to encode your password.')
+            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.', 'Symfony\Component\Security\Core\User\User')
+            ->addOption('empty-salt', null, InputOption::VALUE_NONE, 'Do not generate a salt or let the encoder generate one.')
```

Comment:
@jrobeson are you sure that the `--salt` option is anti DX?

**Scenario:** your database requires 10 bytes salt values.

**Proposal 1**:

Execute: `php app/console security:encode-password raw_password --salt="ae037d6f72"`

**Proposal 2**:

1. Create a new Symfony console command
2. Make this command extend the default SecurityEncodePasswor command
3. Override the `generateSalt()` command and think of a better implementation
4. Execute `php app/console security:encode-password raw_password`